btcstaking-tracker/btcslasher/slasher.go

```diff
@@ -251,24 +256,45 @@ func (bs *BTCSlasher) SlashFinalityProvider(extractedFpBTCSK *btcec.PrivateKey)
 	// Initialize a mutex protected *btcec.PrivateKey
 	safeExtractedFpBTCSK := types.NewPrivateKeyWithMutex(extractedFpBTCSK)
 
+	// Initialize a semaphore to control the number of concurrent operations
+	sem := semaphore.NewWeighted(bs.maxSlashingConcurrency)
+
 	// try to slash both staking and unbonding txs for each BTC delegation
 	// sign and submit slashing tx for each active delegation
-	// TODO: use semaphore to prevent spamming BTC node
 	for _, del := range activeBTCDels {
```

Comment:
just realized that the active BTC dels and unbonded BTC dels are handled exactly the same, suggestion to append both and only pass one `for`

```go
	dels := append(activeBTCDels, unbondedBTCDels...)

	for _, del := range dels {
            ....
        }
```

Reply:
Totally, thanks!